design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,368 @@
+    - mutable: list of packages ready to be upgraded on the Edge Node, 
+      empty if no new versions are available.
+    - immutable: latest EMT version, empty if the version is already latest.
+  - `next_update_policy`: The update policy that is applied during the 
+    maintenance schedule.
+  
+New Resources:
+
+- **OS Update Policy**: defines the Policy that must be applied during an 
+  update. This resource can be linked to multiple Instances, Regions and or 
+  Sites. Some fields and edges are valid for Mutable or Immutable OSes.
+  - `install_packages`: list of packages to be installed during the update. 
+    This field is used only for Mutable OSes.
+  - `update_sources`: 'DEB822 Source Format' for Debian style OSs, contains 
+    the sources of the packages to be installed during the update, if new 
+    sources needs to be added.
+  - `update_policy`: enum field, defines which policy to use:
+    - `UPDATE_POLICY_LATEST`: upgrade to latest version. 
+      - mutable: upgrade all packages to latest versions.
+      - immutable: upgrade the OS version to the latest, based on SemVer.
+    - `UPDATE_POLICY_TARGET`: upgrade to a specific version.
+      - mutable: upgrade all packages to the specified version defined in 
+        the `install_package` list.
+      - immutable: upgrade the OS version to OS Version linked by the 
+        `target_os` edge.
+  - `target_os`: edge to the OS Profile that must be installed during the 
+    update. This field is used only for Immutable OSes, and when the 
+    `update_policy` is `UPDATE_POLICY_TARGET`.
+  
+- **OS Update Run**: stores the information about the update job that run 
+  on the Edge Node. This resource is created when the update is started, 
+  and stores information about what happened in the Edge Node during the 
+  Update. If there are concerns about the size of this table, we can add 
+  retention policies for this resource.
+  - `status`: stores the short description of what happened during the update.
+  - `status_indicator`: enum field, moder status indicator:
+    - `STATUS_INDICATION_ERROR`: an error happened during the update.
+    - `STATUS_INDICATION_IN_PROGRESS`: unused, the progress about the 
+      update is kept into the `update_status` field of the Instance.
+    - `STATUS_INDICATION_IDLE`: the update was successful.
+  - `status_timestamp`: timestamp of the last status update. Overlaps with 
+    the `end_time`, but it's kept for consistency with Modern Status. Also, 
+    could be used in the future, if we deprecate the Instance `update_status`.
+  - `status_details`: detailed description of what happend during the 
+    update. For example, the list of packages that were installed or 
+    updated, or the reason why the Immutable OS update failed.
+  - `start_time`: timestamp of when the update job started.
+  - `end_time`: timestamp of when the update job ended.
+  - `applied_policy`: the policy that was applied during the update. 
```

Comment:
If the policy type is `LATEST`, it may be hard to track what has changed (except for the human-readable status above). Do you think we should add some fields to explicitly track versions/OS that changed or the human-readable status should be enough ?